root/types/functions.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * @flow strict
+ * Copyright (C) 2017 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+/*
+ * Types are in alphabetical order.
+ *
+ * The definitions in this file are intended to model the output of the
+ * TO_JSON methods under lib/MusicBrainz/Server/Entity/, those are precisely
+ * how data is serialized for us.
+ */
+
+declare type AnchorProps = {
+  +className?: string,
+  +href: string,
+  +key?: number | string,
+  +target?: '_blank',
+  +title?: string,
+};
+
+declare type Expand2ReactInput = VarSubstArg | AnchorProps;
```

Comment:
this file seems a bit nebulous to me (most of these aren't functions)

Maybe move KnockoutObservable/KnockoutObservable into a knockout.js and the rest into an i18n.js? StrOrNum is the only one that's really general...kind of a weird case, like PagerT.

Reply:
Moved PagerT and StrOrNum to a "misc" file for now, but yeah, I'm also not sure what to do.